app/common/model/src/main/java/io/syndesis/common/model/DataShape.java

```diff
@@ -162,7 +159,7 @@ public Builder compress() {
                 } catch (IOException e) {
                     throw new IllegalStateException("Failed to compress data shape", e);
                 }
-            } else if (currentShape.getVariants().parallelStream().anyMatch(variant -> "true".equals(variant.getMetadata().get(COMPRESSION_METADATA_KEY)))) {
+            } else if (currentShape.getVariants().parallelStream().anyMatch(variant -> "true".equals(variant.getMetadata().get(DataShapeMetaData.COMPRESSION)))) {
```

Comment:
```suggestion
            } else if (currentShape.getVariants().stream().anyMatch(variant -> Boolean.parseBoolean(variant.getMetadata().get(DataShapeMetaData.COMPRESSION)))) {
```

`parallelStream` should be slower on a small number of elements.